virttest/qemu_devices/qcontainer.py

```diff
@@ -2548,9 +2549,13 @@ def dimm_device_define_by_params(self, params, name):
         params = params.object_params("dimm")
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
         attrs = qdevices.Dimm.__attributes__[:]
-        params = params.copy_from_keys(attrs)
-        dev = qdevices.Dimm(params=params, dimm_type=dimm_type)
+        attrs_params = params.copy_from_keys(attrs)
```

Comment:
Could you please explain why a new variant is needed here?

Reply:
`params.copy_from_keys(attrs)` will redefine the `params` object, then `params.keys()` becomes `qdevices.Dimm.__attributes__[:]`.
We need to get other keys after creating the dimm device, `nvdimm_set_uuid` and `dimm_extra_params` only exist in the first `params`.
Maybe I can modify it more clearly:
```python
        attrs = qdevices.Dimm.__attributes__[:]
        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
                            dimm_type=dimm_type)
```